packages/http-signature-utils/src/utils/key.ts

```diff
@@ -1,44 +1,62 @@
 import * as crypto from 'crypto'
 import * as fs from 'fs'
 
-export function parseOrProvisionKey(
-  keyFile: string | undefined
-): crypto.KeyObject {
-  const TMP_DIR = './tmp'
-  if (keyFile) {
-    try {
-      const key = crypto.createPrivateKey(fs.readFileSync(keyFile))
-      if (checkKey(key)) {
-        console.log(`Key ${keyFile} loaded.`)
-        return key
-      } else {
-        console.log('Private key is not EdDSA-Ed25519 key. Generating new key.')
-      }
-    } catch (err) {
-      console.log('Private key could not be loaded.')
-      throw err
-    }
+export function loadKey(keyFilePath: string): crypto.KeyObject {
+  let fileBuffer
+  try {
+    fileBuffer = fs.readFileSync(keyFilePath)
+  } catch (error) {
+    throw new Error(`Could not load file: ${keyFilePath}`)
+  }
+
+  let key
+  try {
+    key = crypto.createPrivateKey(fileBuffer)
+  } catch (error) {
+    throw new Error('File was loaded, but private key was invalid')
+  }
+
+  if (!isKeyEd25519(key)) {
+    throw new Error('Private key did not have Ed25519 curve')
   }
+
+  return key
+}
+
+export function generateKey(dir: string): crypto.KeyObject {
   const keypair = crypto.generateKeyPairSync('ed25519')
-  if (!fs.existsSync(TMP_DIR)) {
-    fs.mkdirSync(TMP_DIR)
+  if (!fs.existsSync(dir)) {
+    fs.mkdirSync(dir)
   }
   fs.writeFileSync(
-    `${TMP_DIR}/private-key-${new Date().getTime()}.pem`,
```

Comment:
@sabineschaller what was the use case for actually saving the key to a file when calling `parseOrProvisionKey` ?

Reply:
In Rafiki, we call `parseOrProvisionKey` during backend app config, and use the generated KeyObject directly if the file for the key wasn't provided. We don't really end up using the file of the key